modules/realm/freeipa.rb

```diff
@@ -39,17 +40,23 @@ def initialize
       if errors.empty?
         # Get krb5 token
         init_krb5_ccache Proxy::Realm::Plugin.settings.realm_keytab, Proxy::Realm::Plugin.settings.realm_principal
-        gssapi = GSSAPI::Simple.new(@ipa_server.host, "HTTP")
-        token = gssapi.init_context
+        @gssapi = GSSAPI::Simple.new(@ipa_server.host, "HTTP")
+        token = @gssapi.init_context
+
+        login = Net::HTTP.new(@ipa_server.host, 443)
+        login.use_ssl = true
+        login.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+        request = Net::HTTP::Post.new("/ipa/session/login_kerberos", 'Authorization'=>"Negotiate #{strict_encode64(token)}", 'Referer' => @ipa_server.to_s)
```

Comment:
Looks like support for sessions for xml api was introduced in freeipa version 3.0. That login end-point first appeared in version 2.2, I think.